### PR TITLE
Add missing tabs to org projects page

### LINKED ISSUE #22676: org projects page menu is incorrect and have different style of header's background color and url
### Description

org projects page menu is incorrect:
1. build the last version and run gitea server
2. after the init proccess, create a new org
3. access `projects` , `people` and `team` tab is missing

org projects

the right one
org packages

different style of header's background color and url:
| Page | Style | Url |
| ---- | ---- | ---- |
| projects | template/user/overview/header.tmpl | orgname/-/projects |
| package | template/user/overview/header.tmpl | orgname/-/packages |
| people | template/org/header.tmpl | org/orgname/members |
| team | template/org/header.tmpl | org/orgname/teams |

org people
org teams

### Gitea Version

4011821c946e8db032be86266dd9364ccb204118

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build in local

### Database

None

### PR DESCRIPTION
**Pull request by yp05327**
Fixes https://github.com/go-gitea/gitea/issues/22676

Context Data `IsOrganizationMember` and `IsOrganizationOwner` is used to control the visibility of `people` and `team` tab.
https://github.com/go-gitea/gitea/blob/2871ea08096cba15546f357d0ec473734ee9d8be/templates/org/menu.tmpl#L19-L40

And because of the reuse of user projects page, User Context is changed to Organization Context. But the value of  `IsOrganizationMember` and `IsOrganizationOwner` are not being given.

I reused func `HandleOrgAssignment` to add them to the ctx, but may have some unnecessary variables, idk whether it is ok.

I found there is a missing `PageIsViewProjects` at create project page.

### CODE REVIEW

File: modules/context/org.go [hunk 1]
```diff
@@ -77,25 +99,13 @@ func HandleOrgAssignment(ctx *Context, args ...bool) {
...
-				ctx.ServerError("LookupUserRedirect", err)
-			}
-		} else {
-			ctx.ServerError("GetUserByName", err)
-		}
-		return
+
+	// if Organization is not defined, get it from params
+	if ctx.Org.Organization == nil {
+		GetOrganizationByParams(ctx)
```
Comment (lunny) [inline modules/context/org.go]:
```suggestion
		GetOrganizationByParams(ctx)
		if ctx.Written() {
		    return
		}
```

**Comment by yp05327** [comment]
You will see a `package` tab in projects page, even if package is disabled in app.ini.
e.g.
https://gitea.com/gitea/-/projects

Because
https://github.com/go-gitea/gitea/blob/be93e48ccb73acfaa372e56217105d3420e1a40e/templates/user/overview/header.tmpl#L28

`UnitPackagesGlobalDisabled` is not defined at any place.

https://github.com/go-gitea/gitea/blob/be93e48ccb73acfaa372e56217105d3420e1a40e/templates/org/menu.tmpl#L9
I think `IsPackageEnabled` used in org/menu.tmpl is the right one.

Commit https://github.com/go-gitea/gitea/pull/22705/commits/ae2024483c2ff028864ae2c711ed7fd818b4b045 fixes this.

**Comment by yp05327** [comment]
Code page uses `templates/user/overview/header.tmpl` to render the header.
There is no need to decide whether `Code` Tab is in active or not in `org/menu.tmpl`
And `PageIsOrgCode`'s value is not defined at any place.
I think it is better to remove this.

Commit https://github.com/go-gitea/gitea/pull/22705/commits/9e3ed4c7d2ef3885a67c0c82734e7d09085f26b2 fixes this.

**Review by lunny — APPROVED** [review]

**Comment by yp05327** [comment]
I missed checking team permission.
`No Access` has no effect to control the visibility of `Project` `Package` tab. 
I think we can merge this PR first, and fix it in https://github.com/go-gitea/gitea/pull/22865
Because some important changes [here](https://github.com/go-gitea/gitea/pull/22705/files#diff-cdc93195a33eaa58dbab1eaa18552a877f0838a39c7448eb861717f343e696a4R67) are also used in https://github.com/go-gitea/gitea/pull/22865

**Comment by yp05327** [comment]
Maybe we should use `UserAssignments` as `IndividualAssignments`, and do not migrate `OrgAssgnments` into `UserAssignments`. This will cause a clutter design.

Then we can use `?tab=projects` to access individual projects list page by using `UserAssignments`, and use `org/{orgname}/projects` to access organization projects  list page by using `OrgAssgnments` which looks better I think.
I will do this in another PR, and after finish it, I will close this one.

**Comment by yp05327** [comment]
I designed a new solution to avoid the CI error.

**Review by delvh — APPROVED** [review]

**Comment by jolheiser** [comment]
🎺 🤖

**Pull request merged**